### PR TITLE
ISSUE-4076: Use the central get_engine method for usage

### LINKED ISSUE #4076: Clickhouse Usage get_engine
Use the central get_engine method for clickhouse usage and deprecate `_get_connection` from SQLAlchemyHelper

### PR DESCRIPTION
**Pull request by MilanBariya**
### Describe your changes :

I worked on Use the central get_engine method for clickhouseUsage, MssqlUsageSource,  RedshiftUsage, SnowflakeUsage and deprecate _get_connection from SQLAlchemyHelper
fix #4076 

#
### Type of change :

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @ayush-shah @pmbrull

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/mssql_usage.py [hunk 1]
```diff
@@ -17,60 +17,36 @@
...
         self.analysis_date = start
         self.sql_stmt = MSSQL_SQL_USAGE_STATEMENT.format(start_date=start, end_date=end)
-        self.alchemy_helper = SQLAlchemyHelper(
-            self.connection,
-            metadata_config,
-            DatabaseServiceType.Mssql.value,
-            self.sql_stmt,
-        )
         self.report = SQLSourceStatus()
+        self.engine_connecter = get_engine(self.config.serviceConnection)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/mssql_usage.py:46]: Can we keep the name just as engine? TO be consistent with other sources

File: ingestion/src/metadata/ingestion/source/redshift_usage.py [hunk 2]
```diff
@@ -62,15 +61,13 @@ def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnecti
             start_time=start, end_time=end
         )
         self.analysis_date = start
-        self.alchemy_helper = SQLAlchemyHelper(
-            self.service_connection, metadata_config, "Redshift", self.sql_stmt
-        )
         self._extract_iter: Union[None, Iterator] = None
         self._database = "redshift"
         self.status = SQLSourceStatus()
+        self.engine_connecter = get_engine(self.config.serviceConnection)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/redshift_usage.py:67]: same as above, please just name this as engine

File: ingestion/src/metadata/ingestion/source/mssql_usage.py (hunk 1 above)
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/mssql_usage.py:46]: @pmbrull Sure

File: ingestion/src/metadata/ingestion/source/clickhouse_usage.py [hunk 3]
```diff
@@ -130,7 +102,10 @@ def get_report(self):
...
 
     def close(self):
-        self.alchemy_helper.close()
+        pass
 
     def get_status(self) -> SourceStatus:
         return self.report
+
+    def test_connection(self) -> None:
+        pass
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/clickhouse_usage.py:111]: Let's use the same test_connection approach as with the SQLSource, as we are using SQA engines

**Review by pmbrull — APPROVED** [review]
thanks for the changes, LGTM!

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**